homeassistant/components/alarm_control_panel/alarmdecoder.py

```diff
@@ -44,32 +41,28 @@ def __init__(self, name, hass):
         self._name = name
         self._state = STATE_UNKNOWN
 
-        _LOGGER.debug("Setting up panel")
+        dispatcher_connect(
+            hass, SIGNAL_PANEL_MESSAGE, self._message_callback)
 
-    @asyncio.coroutine
-    def async_added_to_hass(self):
-        """Register callbacks."""
-        async_dispatcher_connect(
-            self.hass, SIGNAL_PANEL_MESSAGE, self._message_callback)
+        _LOGGER.debug("Setting up panel")
 
-    @callback
```

Comment:
This callback I think you can keep to be run within the event loop. So revert the changes within it below too.